tests/failpoints_tests.rs

```diff
@@ -0,0 +1,60 @@
+#![cfg(feature = "integration-tests")]
+
+mod common;
+use common::{clear_tikv, pd_addrs};
+use serial_test::serial;
+use tikv_client::{Result, TransactionClient, TransactionOptions};
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 6)]
+#[serial]
+async fn optimistic_heartbeat() -> Result<()> {
```

Comment:
Maybe add a few comments to explain the tests?

Reply:
I think this test would be better off as a unit test - you could use a mock to check the heartbeat message is sent without having to use failpoints and rely on timing.